### DISCUSSION TITLE
Modifying stress induced diffusion equation in Composite electrode model

### DISCUSSION THREAD

**Discussion by MehrabHasan** [pybamm-team/PyBaMM#4510] [Q&A]
Hi,
I am trying to apply the stress-induced diffusion equation to do so I am trying to modify the concentration equations in the Composite electrode source code. But I  have a 2nd order pde term and I cannot include that in the self.rhs. Whenever I try to do so, I get an error that initial_conditions shape and self.rhs shape are not matching. How can I fix that? I am getting the following error:

"ModelError: rhs and initial conditions must have the same shape after discretization but rhs.shape = (420, 1) and initial_conditions.shape = (400, 1) for variable 'Negative primary particle concentration [mol.m-3]'."
Following is the equation, I am trying to model.
[equation.pdf](https://github.com/user-attachments/files/17346983/equation.pdf)

**Comment by DrSOKane** [comment]
Stress-induced diffusion is already included in PyBaMM! It should be applied automatically if the particle mechanics option is "swelling only" or "swelling and cracking".

**Reply by MehrabHasan** [reply]
unfortunately, whenever I try to use that for multiple-phase particles, it does not work.
